jlm/rvsdg/theta.hpp

```diff
@@ -355,6 +355,30 @@ class theta_output final : public structural_output
   jlm::rvsdg::theta_input * input_;
 };
 
+/**
+ * Represents a region argument in a theta subregion.
+ */
+class ThetaArgument final : public argument
+{
+  friend theta_node;
+
+public:
+  ~ThetaArgument() noexcept override;
+
+private:
+  ThetaArgument(rvsdg::region & region, theta_input & input)
+      : argument(&region, &input, input.Type())
+  {}
+
+  static ThetaArgument &
+  Create(rvsdg::region & region, theta_input & input)
```

Comment:
Is there any way of assuring that the region is a theta region?

Reply:
The constructor and creator method of the ThetaArgument class is private and the theta_node class is marked as a friend class. Thus, only the theta_node class is allowed to create a ThetaArgument class. In the theta_node class, the ThetaArgument class is created exactly at one location, and the theta subregion is handed in at that location.

Just to be sure, I added a `JLM_ASSERT()` in the ThetaArgument constructor to check whether it is a theta subregion.